assets/js/modules/search-console/components/ModulePopularPagesWidget.js

```diff
@@ -0,0 +1,167 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x, sprintf, _n } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { numFmt } from '../../../util';
+import { MODULES_SEARCH_CONSOLE, STORE_NAME, DATE_RANGE_OFFSET } from '../datastore/constants';
+import PreviewTable from '../../../components/PreviewTable';
+import Link from '../../../components/Link';
+import { CORE_SITE } from '../../../googlesitekit/datastore/site/constants';
+import { CORE_USER } from '../../../googlesitekit/datastore/user/constants';
+import TableOverflowContainer from '../../../components/TableOverflowContainer';
+import ReportTable from '../../../components/ReportTable';
+import { isZeroReport } from '../util/is-zero-report';
+import { getCurrentDateRangeDayCount } from '../../../util/date-range';
+import { generateDateRangeArgs } from '../util/report-date-range-args';
+import ModuleHeader from './common/ModuleHeader';
+
+const { useSelect } = Data;
+
+function ModulePopularPagesWidget( { Widget, WidgetReportZero, WidgetReportError } ) {
+	const {
+		data,
+		isLoading,
+		error,
+		baseServiceURL,
+		currentDayCount,
+	} = useSelect( ( select ) => {
+		const store = select( STORE_NAME );
+
+		const url = select( CORE_SITE ).getCurrentEntityURL();
+		const dateRangeDates = select( CORE_USER ).getDateRangeDates( { offsetDays: DATE_RANGE_OFFSET } );
+		const dateRange = select( CORE_USER ).getDateRange();
+		const { startDate, endDate } = dateRangeDates;
+
+		const reportArgs = {
+			startDate,
+			endDate,
+			url: url || undefined,
+			dimensions: 'query',
+			limit: 10,
+		};
+
+		return {
+			data: store.getReport( reportArgs ),
+			isLoading: ! store.hasFinishedResolution( 'getReport', [ reportArgs ] ),
+			error: store.getErrorForSelector( 'getReport', [ reportArgs ] ),
+			baseServiceURL: store.getServiceReportURL( {
+				...generateDateRangeArgs( dateRangeDates ),
+				page: url ? `!${ url }` : undefined,
+			} ),
+			currentDayCount: getCurrentDateRangeDayCount( dateRange ),
+		};
+	} );
+
+	if ( isLoading ) {
+		return <PreviewTable padding />;
+	}
+
+	if ( error ) {
+		return <WidgetReportError moduleSlug="search-console" error={ error } />;
+	}
+
+	if ( isZeroReport( data ) ) {
+		return <WidgetReportZero moduleSlug="search-console" />;
+	}
+
+	return (
+		<Widget
+			noPadding
+			Header={ () => (
+				<ModuleHeader
+					title={ sprintf(
+						/* translators: %s: number of days */
+						_n( 'Top search queries over the last %s day', 'Top search queries over last %s days', currentDayCount, 'google-site-kit', ),
+						currentDayCount,
+					) }
+					ctaLink={ baseServiceURL }
+					ctaLabel={ sprintf(
+						/* translators: %s: module name. */
+						__( 'See full stats in %s', 'google-site-kit' ),
+						_x( 'Search Console', 'Service name', 'google-site-kit' )
+					) }
```

Comment:
There is some overlap with #3015 here which introduces the necessary CSS styles and two new `WidgetHeaderTitle` and `WidgetHeaderCTA` components that should be used instead. I already left a similar comment in https://github.com/google/site-kit-wp/pull/2918#discussion_r600648816, so maybe it would be best to take these two components and the related CSS changes out of #3015 and copy them into here, to unblock this.